sycl/test/regression/unnamed-lambda-split-order.cpp

```diff
@@ -0,0 +1,8 @@
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda %s -o %t.out
+#include <sycl.hpp>
```

Comment:
Can you please add a small comment here what this test checks?

Reply:
Done.  Also moved it to unnamed-lambda.cpp instead of its own file, as @AlexeySachkov  requested offline.